classes/coc/script/Eval.as

```diff
@@ -11,49 +11,91 @@ public class Eval {
 			return condition() ? then() : elze();
 		};
 	}
-
-	private var thiz:*;
+	
+	private var scopes:/*Object*/Array;
 	private var expr:String;
-	private var src:String;
+	private var _src:String;
 	private var _call:Function;
-	public function call(thiz:Object):* {
-		this.thiz = thiz;
+	public function call(...scopes:/*Object*/Array):* {
```

Comment:
Please add documentation, so the next person working on it will know what the function does.